gen3-integration-tests/tests/test_gen3_workflows.py

```diff
@@ -0,0 +1,565 @@
+        transient_states = {"QUEUED", "INITIALIZING", "RUNNING"}
+        final_success_state = "COMPLETE"
+        final_failure_states = {"FAILED", "EXECUTOR_ERROR", "CANCELED"}
+
+        for attempt in range(1, max_retries + 1):
+            task_info = self.gen3_workflow.get_tes_task(
+                task_id=task_id,
+                user=self.valid_user,
+                expected_status=200,
+            )
+            state = task_info.get("state")
+
+            if state == final_success_state:
+                logger.info("TES task completed successfully")
+                break
+
+            assert (
+                state not in final_failure_states
+            ), f"TES task failed. Final state: {state}, Response: {task_info}"
+            assert (
+                state in transient_states
+            ), f"Unexpected TES task state '{state}' encountered. Response: {task_info}"
+
+            logger.debug(
+                f"Attempt {attempt} of {max_retries}: Task state is '{state}', retrying..."
+            )
+            time.sleep(poll_interval)
+        else:
+            raise Exception(
+                f"TES task did not complete in time. Final state: {state}, Response: {task_info}"
+            )
+
+        # Step 5: Validate task logs
+        task_logs = task_info.get("logs", [])
+
+        assert (
+            len(task_logs) > 0 and len(task_logs[0].get("logs", [])) > 0
+        ), f"Expected task logs to be present and have at least one log entry, but got: {task_logs}"
+
+        assert (
+            "stdout" in task_logs[0]["logs"][0]
+        ), f"Expected task log entry to have 'stdout', but got: {task_logs[0]['logs'][0]}"
+
+        # Check if the stdout contains the expected echo message
+        stdout = task_logs[0]["logs"][0]["stdout"].strip()
+        assert (
+            stdout == echo_message
+        ), f"Expected stdout to be `Done!`, but found {stdout} instead."
+
+        # Step 6: Validate task outputs
+
+        s3_contents = {}
+        for file_name in ["output.txt", "grep_output.txt"]:
+            response = self.gen3_workflow.get_bucket_object_with_boto3(
+                object_path=f"{s3_path_prefix}/{file_name}",
+                s3_storage_config=self.s3_storage_config,
+                user=self.valid_user,
+                expected_status=200,
+            )
+
+            try:
+                s3_contents[file_name] = response["Body"].read().decode("utf-8")
+            except Exception as e:
+                raise ValueError(
+                    f"Failed to read or decode content of {file_name} from S3. Error: {e}"
+                )
+
+        assert (
+            message in s3_contents["output.txt"]
+        ), f"The output_response of the TES task did not return the expected output. Expected: '{message}' to be in output_response, but found '{s3_contents['output.txt']}' instead."
+
+        assert (
+            message in s3_contents["grep_output.txt"].strip()
```

Comment:
as discussed, we need a TODO here that references the ticket about extra text in the output.